platform/default/src/mbgl/storage/pmtiles_file_source.cpp

```diff
@@ -0,0 +1,547 @@
+#include <sstream>
+#include <map>
+
+#include <mbgl/platform/settings.hpp>
+#include <mbgl/storage/pmtiles_file_source.hpp>
+#include <mbgl/storage/file_source_manager.hpp>
+#include <mbgl/storage/file_source_request.hpp>
+
+#include <rapidjson/document.h>
+#include <rapidjson/rapidjson.h>
+#include <rapidjson/stringbuffer.h>
+#include <rapidjson/writer.h>
+
+#include <mbgl/util/thread.hpp>
+#include <mbgl/util/url.hpp>
+#include <mbgl/util/chrono.hpp>
+#include <mbgl/util/compression.hpp>
+#include <mbgl/util/filesystem.hpp>
+
+#include <pmtiles.hpp>
+
+#include <sys/types.h>
+#include <sys/stat.h>
+
+#if defined(__QT__) && (defined(_WIN32) || defined(__EMSCRIPTEN__))
+#include <QtZlib/zlib.h>
+#else
+#include <zlib.h>
+#endif
+
+namespace {
+const int MAX_DIRECTORY_CACHE_ENTRIES = 100;
+
+bool acceptsURL(const std::string& url) {
+    return 0 == url.rfind(mbgl::util::PMTILES_PROTOCOL, 0);
```

Comment:
It's way better to use `starts_with`, I will change this line. Most of strange/bad code is because I tried to make it alike the `mbtiles_file_source.cpp` as much as I can (in fact, I started with the `mbtiles_file_source.cpp` code, and changed it until I achieved this version of the `pmtiles_file_source.cpp`), but of course, the bad parts will be reworked to produce a better and more readable code.

Reply:
Done!